src/Worker.Extensions.DurableTask/FunctionsDurableClientProvider.cs

```diff
@@ -0,0 +1,128 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.DurableTask.Client;
+using Microsoft.DurableTask.Client.Grpc;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Azure.Functions.Worker.Extensions.DurableTask;
+
+/// <summary>
+/// The functions implementation of the durable task client provider.
+/// </summary>
+internal class FunctionsDurableClientProvider : IDurableTaskClientProvider, IAsyncDisposable
```

Comment:
I am re-using a public contract from the durable-dotnet SDK. I am not sure if I want to stick with that or roll my own contract. I don't want anyone taking a direct dependency on this.